tests/apps/test_osticket.py

```diff
@@ -0,0 +1,30 @@
+""" Test the osticket app """
+
+import logging
+
+import pytest
+
+from cg.apps.osticket import OsTicket
+from cg.exc import TicketCreationError
+
+
+def test_osticket_respone_500(monkeypatch, caplog, response):
+    """Test that we log properly when we get a non successful response"""
+
+    # GIVEN a ticket server always gives a failure in response
+    osticket_api = OsTicket()
+    monkeypatch.setattr("requests.post", lambda *args, **kwargs: response)
```

Comment:
Can you explain what's happening here?

Reply:
Yes, when post is called on the requests API we return our handcrafted response that will provoke the behaviour we want to test

Reply:
:clock1: tik tok .. now I get it! The lambda is to 'catch the post'.

Disclaimer: I've only used monkeypatch with a pass-on function, not an anonymous function. As in:

```
    def mock_get(*args, **kwargs):
        return MockResponse()

    monkeypatch.setattr(requests, "get", mock_get)
```

So, I guess how you solved it here was to avoid the anonymous function, right?

Reply:
Maybe you could say it like that, I'm thinking of it as "connect first part (function post on requests) to a new action, our lambda expression".

So I'd say it's more like using another feature of setattr on monkeypatch, you can give the whole thing to catch as the first argument:
```    
    def mock_get(*args, **kwargs):
        return MockResponse()

    monkeypatch.setattr("requests.get", mock_get)
```

From monkeypatch.py:
        _For convenience you can specify a string as ``target`` which
        will be interpreted as a dotted import path, with the last part
        being the attribute name.  Example:
        ``monkeypatch.setattr("os.getcwd", lambda: "/")``
        would set the ``getcwd`` function of the ``os`` module._

Reply:
Sweeet :)